agent/src/podman/workload.rs

```diff
@@ -156,7 +157,8 @@ impl PodmanWorkload {
                     workload_name: self.workload_spec.workload.name.to_string(),
                     execution_state: ExecutionState::ExecFailed,
                 }])
-                .await;
+                .await
+                .unwrap_or_exit("Could not update workload state after failed container start.");
```

Comment:
I have added `unwrap_or_unreachable` to Option and Result. I have also restructured the std extensions into a separate folder. Feel free to move the extension traits also into a separate folder if you think this would have a benefit.